crates/sargon/src/system/sargon_os/sargon_os_sync_accounts.rs

```diff
@@ -0,0 +1,52 @@
+use crate::prelude::*;
+
+// ==================
+// Sync Profile Accounts with status on ledger
+// ==================
+impl SargonOS {
+    /// Checks all active accounts in current network on ledger, if any of them are deleted.
+    /// Any deleted account is marked as tombstoned in profile.
+    ///
+    /// Returns true if any account became tombstoned.
+    pub async fn sync_accounts_deleted_on_ledger(&self) -> Result<bool> {
+        let accounts = self.accounts_on_current_network()?;
+
+        let network_id = self.profile_state_holder.current_network_id()?;
+        let gateway_client = GatewayClient::new(
+            self.clients.http_client.driver.clone(),
+            network_id,
+        );
+
+        let active_account_addresses =
+            accounts.iter().map(|a| a.address).collect::<Vec<_>>();
+        let on_ledger_checks = active_account_addresses
+            .iter()
+            .map(|address| {
+                gateway_client.check_account_is_deleted(address.clone())
```

Comment:
this does `N` many requests. Dont we wanna do **one single** batch request instead?

Reply:
I really think we do, imagine 100 accounts...

Reply:
How? Gateway does not allow this...

Reply:
Not possible with the current gateway endpoint. We could maybe batch those requests ourselves.

Reply:
awaiting https://rdxworks.slack.com/archives/C06EBEA0SGY/p1731673177430949?thread_ts=1731668714.440949&cid=C06EBEA0SGY